### PR TITLE
fix parenthesis encoding in unicode highlighting

### LINKED ISSUE #146225: Incorrect formatting in unicode highlighting
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.65.2
- OS Version: Darwin x64 20.6.0

Steps to Reproduce:

1. Input  fullwidth right parenthesis `（淡）x`
2. Hover the right parenthesis

image

### PR DESCRIPTION
**Pull request by ambar**
This PR fixes #146225

### CODE REVIEW

File: src/vs/editor/contrib/unicodeHighlighter/browser/unicodeHighlighter.ts [hunk 1]
```diff
@@ -510,6 +510,10 @@ export class UnicodeHighlighterHoverParticipant implements IEditorHoverParticipa
 	}
 }
 
+function fixedEncodeURIComponent(str: string): string {
+	return encodeURIComponent(str.replace(/[()]/g, escape));
```
Comment (hediet) [inline src/vs/editor/contrib/unicodeHighlighter/browser/unicodeHighlighter.ts:514]: This double-escapes the string though.

image

Comment (ambar) [reply src/vs/editor/contrib/unicodeHighlighter/browser/unicodeHighlighter.ts]: Updated.

File: src/vs/editor/contrib/unicodeHighlighter/browser/unicodeHighlighter.ts [hunk 2]
```diff
@@ -496,7 +496,7 @@ export class UnicodeHighlighterHoverParticipant implements IEditorHoverParticipa
 
 			const adjustSettings = nls.localize('unicodeHighlight.adjustSettings', 'Adjust settings');
 			const contents: Array<IMarkdownString> = [{
-				value: `${reason} [${adjustSettings}](command:${ShowExcludeOptions.ID}?${encodeURIComponent(JSON.stringify(adjustSettingsArgs))})`,
+				value: `${reason} [${adjustSettings}](command:${ShowExcludeOptions.ID}?${fixedEncodeURIComponent(JSON.stringify(adjustSettingsArgs))})`,
```
Comment (hediet) [inline src/vs/editor/contrib/unicodeHighlighter/browser/unicodeHighlighter.ts:499]: Hmm, the issue actually is that the escaping conflicts with markdown.
I think the correct fix would be to use this helper:

https://github.com/microsoft/vscode/blob/30e1c0cfdaa3d1b3e7d4a0141bb8b0d3bcb51887/src/vs/base/common/htmlContent.ts#L78

Sorry for only noticing now.

Comment (ambar) [reply src/vs/editor/contrib/unicodeHighlighter/browser/unicodeHighlighter.ts]: Replaced with `MarkdownString`

**Comment by hediet** [comment]
Thanks a lot!

**Pull request merged**